code/game/world.dm

```diff
@@ -164,10 +166,30 @@ GLOBAL_VAR(restart_counter)
 	log_runtime(GLOB.revdata.get_log_message())
 
 /world/Topic(T, addr, master, key)
-	TGS_TOPIC	//redirect to server tools if necessary
-
+	var/static/list/bannedsourceaddrs = list()
+	var/static/list/lasttimeaddr = list()
 	var/static/list/topic_handlers = TopicHandlers()
 
+	//LEAVE THIS COOLDOWN HANDLING IN PLACE, OR SO HELP ME I WILL MAKE YOU SUFFER
+	if (addr in bannedsourceaddrs)
+		return
+
+	if (length(T) >= MAX_TOPIC_LEN)
+		log_admin_private("[addr] banned from topic calls for a round for too long status message")
+		bannedsourceaddrs[addr] = TOPIC_BANNED
+		return
+	if (addr in lasttimeaddr)
+		var/lasttime = lasttimeaddr[addr]
+		if(world.time < (lasttime + 2 SECONDS))
+			log_admin_private("[addr] banned from topic calls for a round for too frequent messages")
+			bannedsourceaddrs[addr] = TOPIC_BANNED
+			return
+
+	lasttimeaddr[addr] = world.time

```

Comment:
instead use 
`
lasttimeaddr[addr] = world.time + 2 SECONDS
`
and check that 
`
if(world.time < lasstime)
`